custom profile fields: Add copy link for long text fields on user cards.

Hell1213 opened the pull request:
Long text custom profile fields can now be configured to display on user cards. When enabled, the user card shows the field name as a clickable link with an id-card icon, and a copy icon aligned to the right on the same line. Clicking the field name navigates to the full user profile, while clicking the copy icon copies the full field value to the clipboard. The field value itself is not displayed on the card, only the field name.

The field is only displayed on the user card when the value is non-empty and the `display_on_user_card` option is enabled in the admin settings.

The `display_on_user_card` option is stored in the `field_data` JSON for LONG_TEXT type fields and is only available for this field type.

**Implementation details:**

- Added admin UI checkbox "Display link on user card" for LONG_TEXT fields only
- Field name is shown on user card as a clickable link with id-card icon (zulip-icon-id-card)
- Copy icon appears on the same line, aligned to the right
- Clicking the field name navigates to the full user profile via `view_full_user_profile`
- Copy functionality uses `navigator.clipboard.writeText()` with confirmation tooltip
- Backend handles `field_data` serialization for LONG_TEXT fields in create/update operations
- CSS improvements: Flexbox layout keeps icon, title, and copy icon on same line with unified hover states
- Fixed TypeScript errors by using bracket notation for index signature property access

Fixes: #36517

**How changes were tested:**

- All backend tests in `zerver/tests/test_custom_profile_data.py` pass
- All linters pass (ESLint, Prettier, TypeScript type checking, Ruff, thirdparty)
- Manual testing:
  - Created a long text custom profile field
  - Enabled "Display link on user card" checkbox in admin settings
  - Verified field name appears on user card with id-card icon and copy icon on the same line
  - Tested clicking field name - successfully navigates to full user profile
  - Tested copy functionality - successfully copies full text to clipboard
  - Verified field only appears when value is non-empty
  - Tested toggle functionality in admin table

<details>

<summary>Self-review checklist</summary>

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability (variable names, code reuse, readability, etc.).

**Communicate decisions, questions, and potential concerns:**

- [x] Explains differences from previous plans (e.g., issue description).

  - Field name shown as clickable link with id-card icon (not just text)
  - Clicking field name navigates to full user profile
  - Copy icon aligned to the right on the same line
  - Only displays when value is non-empty

- [x] Highlights technical choices and bugs encountered.

  - Used `field_data` JSON to store `display_on_user_card` option (string "true" format)
  - Fixed TypeScript TS4111 errors by using bracket notation for index signature property access (`z.record()` schemas)
  - Used flexbox layout to keep icon, title, and copy icon on same line
  - Added id-card.svg icon to THIRDPARTY file in correct Debian copyright format

- [ ] Calls out remaining decisions and concerns.

  - None

- [x] Automated tests verify logic where appropriate.
  - All backend tests pass, including field_data handling for LONG_TEXT fields

**Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)):**

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

**Completed manual review and testing of the following:**

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.

</details>

** Screenshots**

<img width="382" height="553" alt="Screenshot from 2026-01-07 00-39-21" src="https://github.com/user-attachments/assets/cae64e61-52e7-49a4-8d0a-f7b47afb8ed8" />
<img width="382" height="553" alt="Screenshot from 2026-01-07 00-39-31" src="https://github.com/user-attachments/assets/225ecf87-0326-4ba1-a1d0-17b14db834db" />

<img width="695" height="527" alt="Screenshot from 2026-01-04 22-58-13" src="https://github.com/user-attachments/assets/3499ebb0-fc5b-42de-bc3b-eb35bd028df6" />
<img width="695" height="527" alt="Screenshot from 2026-01-04 22-58-07" src="https://github.com/user-attachments/assets/d9bbbba9-72cd-4b35-9bce-8a18a5b68225" />



**Screenshots of Email field copy button icon for test only ** : 

<img width="382" height="553" alt="Screenshot from 2026-01-06 17-25-53" src="https://github.com/user-attachments/assets/82a875b7-6c9d-4eb4-8810-0ef1edc4be2d" />
<img width="382" height="553" alt="Screenshot from 2026-01-06 17-25-58" src="https://github.com/user-attachments/assets/034ee438-fe65-4c53-b63e-6e5e0126e6b4" />

alya commented:
Thanks! I posted some thoughts in [#issues > display long custom profile field @ 💬](https://chat.zulip.org/#narrow/channel/9-issues/topic/display.20long.20custom.20profile.20field/near/2303930); I think we need to updated the spec.

Hell1213 committed:
custom profile fields: Add copy link for long text fields on user cards.

Long text custom profile fields can now be configured to display on
user cards. When enabled, the user card shows the field name as a
clickable link with an id-card icon, and a copy icon aligned to the
right. Clicking the field name navigates to the full user profile,
while clicking the copy icon copies the full field value to the
clipboard.

The field is only displayed when the value is non-empty and the
display_on_user_card option is enabled in admin settings.

- Replace document icon with proper ID card design in id-card.svg
- Fix copy button alignment between email and custom profile fields
- Add copy functionality using async/await pattern for ESLint compliance
- Remove popover-menu-link class from copy button for proper positioning
- Add margin-right adjustment for precise horizontal alignment

Fixes: #36517

Hell1213 commented:
> Please add updated screenshots to the PR description when ready.


thanks . I'm working on it will fix the alignment of copy icon and update the icon as well and when ready will push the changes .!

alya commented:
You should also consider what information in your super-long commit message actually needs to be there.

alya commented:
The new icon also looks vertically misaligned (too high) vs. the option text.

Hell1213 commented:
> The new icon also looks vertically misaligned (too high) vs. the option text.
> You should also consider what information in your super-long commit message actually needs to be there.
> Please add a screenshot showing a "copy" button both on a long email and a custom profile field on the user card, to demonstrate that they match and are properly aligned.

Thanks for the feedback! I'll add a screenshot showing the copy button alignment between email and custom profile fields, fix the icon vertical alignment issue, and clean up the commit message to be more concise. Working on these fixes now.

Hell1213 commented:
> The new icon also looks vertically misaligned (too high) vs. the option text.

PR is ready plss look when you get time . 
updated commit message , alignment of id-card icon and fixed position of copy button (as Email field ).

alya commented:
Hmm, I don't know what you're doing for the icon, but it does *not* look like https://lucide.dev/icons/id-card.

Please keep in mind that if we can't rely on you to carefully implement issues as described and clearly flag any discrepancies, we'll unfortunately be unable to review your work.
